opencti-platform/opencti-graphql/src/utils/refang.js

```diff
@@ -0,0 +1,116 @@
+// Utility to refang defanged IOCs (Indicators of Compromise)
+// Supports common defanging techniques: [.] -> ., hxxp -> http, etc.
+
+import { URL } from 'url';
+
+/**
+ * Refang, clean, normalize, and optionally truncate an indicator value for Defender API submission.
+ * - Refangs common obfuscations
+ * - Extracts and sanitizes URLs
+ * - Encodes path/query safely
+ * - Strips trailing garbage and punctuation
+ * @param {string} input - The potentially defanged string.
+ * @returns {string} - The refanged string.
+ */
+export function refang(input) {
+  if (!input || typeof input !== 'string') return input;
+
+  // Trim first
+  let output = input.trim();
+
+  // Normalize Unicode (NFKC)
+  if (typeof output.normalize === 'function') {
```

Comment:
As ouput is a string, normalize will always be available.
But not sure why its needed to normalize the unicode.
As search is for me an UTF-8, i dont really understand why we need to handle unicode